src/Npgsql/Internal/NpgsqlReadBuffer.cs

```diff
@@ -233,7 +233,7 @@ internal Task Ensure(int count, bool async, bool readingNotifications)
                             // TODO: As an optimization, we can still attempt to send a cancellation request, but after
                             // that immediately break the connection
                             if (connector.AttemptPostgresCancellation &&
-                                !connector.PostgresCancellationPerformed &&
+                                !connector.InternalCancellationRequested &&
```

Comment:
Does this mean we now want to go into PerformPostgresException if no internal cancellation was requested, but a user cancellation was already requested?